packages/repository/src/repositories/relation.factory.ts

```diff
@@ -35,18 +36,28 @@ export interface HasManyDefinition extends RelationDefinitionBase {
  * relation attached to a datasource.
  *
  */
-export function hasManyRepositoryFactory<SourceID, T extends Entity, ID>(
-  sourceModelId: SourceID,
-  relationMetadata: HasManyDefinition,
-  targetRepository: EntityCrudRepository<T, ID>,
-): HasManyEntityCrudRepository<T> {
-  switch (relationMetadata.type) {
-    case RelationType.hasMany:
-      const fkConstraint = {[relationMetadata.keyTo]: sourceModelId};
 
-      return new DefaultHasManyEntityCrudRepository<
-        T,
-        EntityCrudRepository<T, ID>
-      >(targetRepository, fkConstraint);
-  }
+export function createHasManyRepositoryFactory<
+  S extends Entity,
+  T extends Entity
+>(
+  sourceModel: Class<S>,
+  targetRepo: EntityCrudRepository<T, typeof Entity.prototype.id>,
+) {
+  return function(constraint: Partial<S>) {
+    const relationsMeta = sourceModel.relations as RelationMap;
+    for (const property in relationsMeta) {
+      const meta = relationsMeta[property];
+      switch (meta.type) {
+        case RelationType.hasMany:
```

Comment:
IIUC, this code is looking up the first "HasMany" relation defined on the model. Here is what I don't understand - how are we going to support the case where a single model have multiple different HasMany relations?

For example, imagine a gallery site, where each Person owns (has many) Paintings via `painting.ownerId` property and also authored (has many) Paintings via `painting.authorId` property.

- To obtain the correct relation metadata, we need to know the relation name (owner vs author in the example above).
- IMO, it's not the job of `createHasManyRepositoryFactory` to figure out the configuration of the relation. Instead, it's the responsibility of the caller to provide that metadata. 

```ts
class PersonRepository {
  constructor(...) {
    // ...
    this.paintingsOwned = this._createHasManyRepositoryFactoryFor('paintingsOwned');
    this.paintingsAuthored = this._createHasManyRepositoryFactoryFor('paintingsAuthored');
  }
}

// in @loopback/repository
class DefaultEntityCrudRepository<...> {
  // ...
  protected _createHasManyRepositoryFactoryFor<Target>(relationName) {
    const meta = // obtain relation config for "relationName"
    return createHasManyRepositoryFactory<typeof this, Target>(...);
  }
}
```

Regardless of the implementation details/design we eventually choose, please add a unit or an integration test to verify the scenario I described in this comment.

Reply:
@bajtos We like this UX and have applied your feedback. @raymondfeng thoughts on this approach?